crates/store/re_log_types/src/data_source_message.rs

```diff
@@ -0,0 +1,54 @@
+// TODO(andreas): Conceptually these should go to `re_data_source`.
+// However, `re_data_source` depends on everything that _implements_ a datasource, therefore we would get a circular dependency!
+
+use crate::{AbsoluteTimeRange, LogMsg, StoreId, TimelineName};
+
+/// Message from a data source.
+///
+/// May contain limited UI commands for instrumenting the state of the receiving end.
+#[derive(Clone, Debug)]
+pub enum DataSourceMessage {
+    /// See [`LogMsg`].
+    LogMsg(LogMsg),
+
+    /// A UI command that has to be sorted relative to `LogMsg`s.
+    ///
+    /// Non-ui receivers can safely ignore these.
+    UiCommand(DataSourceUiCommand),
+}
+
+impl From<LogMsg> for DataSourceMessage {
+    #[inline]
+    fn from(msg: LogMsg) -> Self {
+        Self::LogMsg(msg)
+    }
+}
+
+impl From<DataSourceUiCommand> for DataSourceMessage {
+    #[inline]
+    fn from(cmd: DataSourceUiCommand) -> Self {
+        Self::UiCommand(cmd)
+    }
+}
+
+/// UI commands issued when streaming in datasets.
+///
+/// If you're not in a ui context you can safely ignore these.
+#[derive(Clone, Debug)]
+pub enum DataSourceUiCommand {
+    AddValidTimeRange {
+        store_id: StoreId,
+
+        /// If `None`, signals that all timelines are entirely valid.
+        timeline: Option<TimelineName>,
+        time_range: AbsoluteTimeRange,
+    },
```

Comment:
Docstring `AddValidTimeRange`!

Also, I suggest breaking out as `struct AddValidTimeRange`

Reply:
went with splitting out as a struct for a bit but decided against now, it just makes all the existing usages worse as there's no such thing as double-into: there's no way to generically implement an `into`/`from` on the split out struct such that it produces `DataSourceMessage` directly